plugins/scanners/scanoss/src/main/kotlin/ScanOssResultParser.kt

```diff
@@ -56,16 +63,23 @@ internal fun generateSummary(startTime: Instant, endTime: Instant, results: List
                 }
 
                 MatchType.snippet -> {
-                    val file = requireNotNull(details.file)
-                    val lines = requireNotNull(details.lines)
-                    val sourceLocations = convertLines(file, lines)
-                    val snippets = getSnippets(details)
-
-                    snippets.forEach { snippet ->
-                        sourceLocations.forEach { sourceLocation ->
-                            // TODO: Aggregate the snippet by source file location.
-                            snippetFindings += SnippetFinding(sourceLocation, setOf(snippet))
+                    val file = requireNotNull(result.filePath)
+                    if (details.status == StatusType.pending) {
+                        val lines = requireNotNull(details.lines)
+                        val sourceLocations = convertLines(file, lines)
+                        val snippets = getSnippets(details)
+
+                        if (sourceLocations.size != snippets.size) {
+                            logger.warn("number of local line ranges does not match with oss lines on file '$file'")
+                        }
+
+                        for (i in 0 until min(sourceLocations.size, snippets.size)) {
+                            snippetFindings += SnippetFinding(sourceLocations[i], setOf(snippets[i]))
                         }
+                    } else {
+                        logger.warn("file '$file' is identified, not including on snippet findings")
```

Comment:
Also log messages should be properly capitalized and end with a dot, like sentences written in a book.